include/TimeLineWidget.h

```diff
@@ -232,12 +172,9 @@ public slots:
 	float m_ppb;
 	float m_snapSize;
 	Song::PlayPos & m_pos;
+	Timeline* m_timeline;
```

Comment:
This isn't an owning pointer - the timeline itself is a member of `Song`, so mustn't be freed separately.

Reply:
Should we use `std::shared_ptr` then?

Reply:
There's nothing inherently wrong with using a raw pointer if it's non owning (ignoring Qt shenanigans).

Reply:
Okay just making sure! I didn't realize the guidelines still had some raw pointer and reference rules. I thought it was preferred to start moving towards the smart pointers for everything.

Reply:
Smart pointers are used to manage memory automatically, i.e. they function as a replacement for manually writing `delete`, `delete[]`, `->release()`, etc. If you wouldn't have written one of those anyway, then you don't need a smart pointer. The timeline itself is not separately allocated on the heap, rather its memory exists as part of that of `Song`. As such, it will be destroyed automatically once `Song`'s destructor runs, and its memory will be freed when the `Song` instance is freed.

There are a couple of ways it would be possible to use `std::shared_ptr` here. One would be to provide a custom deleter that does nothing - this would work identically to a raw pointer but with more overhead, so there would be no point doing so. The other would be to allocate each timeline on the heap using `std::make_shared`, and then have `Song` return copies of the shared pointer. This is conceptually different - instead of `Song` owning the timeline, and `TimeLineWidget` holding a non-owning pointer to it, they would instead share ownership (hence the name `shared_ptr`). I don't think this is the approach we want to take - a view doesn't own the model it views, rather the models form a tree structure which the views observe. We should ensure the model continues to exist while there's a view viewing it, rather than allowing the view to keep alive a zombie model just for the sake of being able to view something.